controllers/mentor.go

```diff
@@ -7,40 +7,115 @@ import (
 	"net/http"
 	"os"
 	"strings"
+	"fmt"
 
 	"kwoc20-backend/models"
+	logs "kwoc20-backend/utils/logs/pkg"
 
 	"github.com/jinzhu/gorm"
+	"github.com/go-kit/kit/log/level"
 )
 
-// MentorOauth Handler for Github OAuth of Mentor
+// MentorOAuth Handler for Github OAuth of Mentor
 func MentorOAuth(w http.ResponseWriter, r *http.Request) {
 	// get the code from frontend
 	var mentorOAuth1 interface{}
-	body, _ := ioutil.ReadAll(r.Body)
-	err := json.Unmarshal(body, &mentorOAuth1)
-	mentorOAuth, _ := mentorOAuth1.(map[string]interface{})
+	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte(`{"message": "` + err.Error() + `"}`))
+		_, writeErr := w.Write([]byte(`{"message": "` + err.Error() + `"}`))
+		if writeErr != nil {
+			logErr := level.Warn(logs.Logger).Log("error", fmt.Sprintf("%v",writeErr))
+			if logErr != nil {
+				panic("Log Error")
+			}
+		}
+		logErr := level.Error(logs.Logger).Log("error", fmt.Sprintf("%v",err))
```

Comment:
Uhh I would rather skip the return value of the logging, as it's generally skipped and not checked. This might sound a bit contradicting, but the return value shouldn't be checked 😄